internal/stats/course.go

```diff
@@ -12,9 +12,12 @@ type CourseMetricType string
 const (
 	CourseMetricTypeUnknown          CourseMetricType = ""
 	CourseMetricTypeGradingTime                       = "grading-time"
+	CourseMetricTypeTaskDuration                      = "task-duration"
 	CourseMetricTypeCodeAnalysisTime                  = "code-analysis-time"
 )
 
+const TASK_KEY = "task"
```

Comment:
Since you have a const for this, you should make one for "anslysis-type" (in internal/analysis/common.go) as well.

Then you can also have a more descriptive name like "ATTRIBUTE_KEY_TASK" (and a similar name for the analysis type).